elastic/kibana
# Use react query for loading preview chart data

Resolves #152438

## 📝 Summary

In this PR, I used react query to improve the loading of preview and metric explorer charts. Also, I used the `EuiLoadingChart` instead of showing a text.

### Before

https://user-images.githubusercontent.com/12370520/224107429-16bd6f1c-a1b7-429c-b56e-e2201f7d767c.mov


### After

https://user-images.githubusercontent.com/12370520/224107649-c8b5c9a3-d371-405d-a0eb-8c1151047437.mov


## 🧪 How to test
- Open a metric threshold/ inventory rule creation flyout and check the preview charts
  - They should work as before
- Check the metric explorer page charts
  - Adding a filter, changing the time range, or adding a group should work as before
 

https://user-images.githubusercontent.com/12370520/224105438-0ffbd1e4-464f-4438-9894-939f5768c4f4.mov
